rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -0,0 +1,141 @@
+
+TRANSFER_FROM_L1_COMPLETED = b'2\tX\x17i0\x80N\xb6l#C\xc74?\xc06}\xc1bIY\x0c\x0f\x19W\x83\xbe\xe1\x99\xd0\x94'  # noqa: E501
+WITHDRAWAL_BONDED = b'\x0c=%\x0cx1\x05\x1ex\xaajVg\x9eY\x03t\xc7\xc4$A_\xfeJ\xa4tI\x1d\xef/\xe7\x05'  # noqa: E501
+
+
+class HopCommonDecoder(DecoderInterface):
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            bridges: dict[ChecksumEvmAddress, HopBridge],
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.bridges = bridges
+        self.bridge_contracts = {bridge.address for bridge in self.bridges.values()}
+
+    def _decode_bridge(
+            self,
+            transaction: 'EvmTransaction',
+            decoded_events: list['EvmEvent'],
+            all_logs: list['EvmTxReceiptLog'],
+    ) -> list['EvmEvent']:
+        """This function is used to decode the brdiging events done via Hop. It checks if already
+        decoded events interacted with the Hop Protocol, and overwrites them if they did."""
+        for _log in all_logs:
+            if _log.topics[0] in WITHDRAWAL_BONDED:
+                self._decode_withdrawal_bonded(context=DecoderContext(
+                    tx_log=_log,
+                    transaction=transaction,
+                    action_items=[],
+                    all_logs=all_logs,
+                    decoded_events=decoded_events,
+                ))
+        return decoded_events
+
+    def _decode_withdrawal_bonded(self, context: DecoderContext) -> DecodingOutput:
+        recipient = hex_or_bytes_to_address(context.transaction.input_data[4:36])
+        if not self.base.is_tracked(recipient):
+            return DEFAULT_DECODING_OUTPUT
+
+        bridge = self.bridges.get(context.tx_log.address)
+        if bridge is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        if bridge.asset == A_ETH:
+            normalized_amount = token_normalized_value_decimals(amount_raw, 18)
+        else:
+            normalized_amount = token_normalized_value(amount_raw, bridge.asset.resolve_to_evm_token())  # noqa: E501
+
+        for event in context.decoded_events:
+            if (
+                event.address in [bridge.amm_wrapper, bridge.address] and
+                event.asset == bridge.asset and
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.counterparty = CPT_HOP
+                event.notes = f'Bridge {event.balance.amount} {event.asset.symbol_or_name()} via hop protocol'  # noqa: E501
```

Comment:
Is the chain you are bridging to information available?

Reply:
No, I don't think it's there.